### ISSUE TITLE
Routing is not updated while the game is paused (was: Monorail path-finding buggy)

### ISSUE THREAD

**Issue by VictorPhilipp** [VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition#161]
Probably caused by the recent changes in TransportLineAI. Users report that path-finding may yield weird results for monorail transport lines.

**Comment by VictorPhilipp** [comment]
Could not reproduce the issue.
https://steamcommunity.com/workshop/filedetails/discussion/583429740/1694922980042698363/

**Comment by aubergine10** [comment]
Just about to go in game, will see if I can reproduce.

**Comment by VictorPhilipp** [comment]
I could finally reproduce the problem: The bug occurs when u-turns are disallowed.
[QuickSave20180602163216655 test.zip](https://github.com/VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition/files/2066365/QuickSave20180602163216655.test.zip)

**Comment by aubergine10** [comment]
Could this also be affecting train routes? I'd previously noticed that external passenger trains visiting a station were despawning rather than leaving the city via going back the way they came (although not checked to see if that is still happening, I've made changes to my rail network since then so will retest).

Rough schematic of the rail setup at time when I noticed the issue:

```
----E----L...
```

Where `---` is rail line, `E` is station that accepts external connections, `L...` is rest of my rail network that only accepts local connections (and does not have any other outside connections except for that first `E` station).

The savegame linked in #140 probably still has that setup (station at Lakeside Services), although ISTR that I'd added stop signals to my rail network by that point which may or may not have resolved the issue.
